src/option.rs

```diff
@@ -95,51 +96,64 @@ impl<T: Miniconf> Miniconf for Option<T> {
         self.0
             .as_ref()
             .map(|value| value.next_path(state, path))
-            .unwrap_or(false)
+            .unwrap_or_default()
     }
 }
 
 impl<T: crate::Serialize + crate::DeserializeOwned> Miniconf for core::option::Option<T> {
-    fn set_path(
+    fn set_path<'a, P: Peekable<Item = &'a str>>(
         &mut self,
-        mut path_parts: core::iter::Peekable<core::str::Split<char>>,
+        path_parts: &mut P,
         value: &[u8],
-    ) -> Result<(), Error> {
+    ) -> Result<usize, Error> {
+        if self.is_none() {
+            return Err(Error::PathNotFound);
+        }
+
         if path_parts.peek().is_some() {
             return Err(Error::PathTooLong);
         }
 
-        *self = serde_json_core::from_slice(value)?.0;
-        Ok(())
+        let (value, len) = serde_json_core::from_slice(value)?;
+        *self = Some(value);
+        Ok(len)
     }
 
-    fn get_path(
+    fn get_path<'a, P: Peekable<Item = &'a str>>(
         &self,
-        mut path_parts: core::iter::Peekable<core::str::Split<char>>,
+        path_parts: &mut P,
         value: &mut [u8],
     ) -> Result<usize, Error> {
+        if self.is_none() {
+            return Err(Error::PathNotFound);
+        }
+
         if path_parts.peek().is_some() {
             return Err(Error::PathTooLong);
         }
 
-        serde_json_core::to_slice(self, value).map_err(|_| Error::SerializationFailed)
+        // Note(unwrap): checked above
+        serde_json_core::to_slice(self.as_ref().unwrap(), value)
```

Comment:
```suggestion
        let data = self.as_ref().ok_or(Error::PathNotFound)?;
        serde_json_core::to_slice(data, value)
```

I think we can get rid of the unwrap and note entirely.

Reply:
But it is already checked before in L120. And we want that for error consistency (not found before too long).

Reply:
OTOH might actually be better to do it that way and add a specific `PathAbsent` error for runtime absence.

Reply:
My point being we can avoid an unwrap. Feel free to put the `as_ref().ok_or()` earlier in the function where it is right now

Reply:
Agreed. The unwrap is gone now.